src/odemis/driver/zeiss.py

```diff
@@ -513,66 +572,99 @@ def _refreshPosition(self):
 
     def _doMoveRel(self, future, shift):
         """
-        move by the shift
-        shift (float): unit m
+        prepare the stage for a relative move using the values in shift
+        :param future: a future that can be used to manage a move
+        :param shift (dict): contains float position values in m or deg
         """
-        x, y, z, _ = self.parent.GetStagePosition()
+        if set(self.axes.keys()).intersection({"rx", "rz", "m"}):
+            move_list = list(self.parent.GetStagePosition6())
+        else:
+            move_list = list(self.parent.GetStagePosition())
+
+        target_pos = {}
+
         if "x" in shift:
-            x += shift["x"] * 1e3
+            move_list[0] += shift["x"] * 1e3
+            target_pos["x"] = move_list[0] * 1e-3
         if "y" in shift:
-            y += shift["y"] * 1e3
+            move_list[1] += shift["y"] * 1e3
+            target_pos["y"] = move_list[1] * 1e-3
         if "z" in shift:
-            z += shift["z"] * 1e3
-
-        target_pos = self._applyInversion({"x": x * 1e-3, "y": y * 1e-3, "z": z * 1e-3})
-        # Check range (for the axes we are moving)
+            move_list[2] += shift["z"] * 1e3
+            target_pos["z"] = move_list[2] * 1e-3
+        if "rx" in shift:
+            move_list[3] += math.degrees(shift["rx"])
+            target_pos["rx"] = move_list[3]
+        if "rz" in shift:
+            move_list[4] += math.degrees(shift["rz"])
+            target_pos["rz"] = move_list[4]
+        if "m" in shift:
+            move_list[5] += shift["m"] * 1e3
+            target_pos["m"] = move_list[5] * 1e-3
+
+        target_pos = self._applyInversion(target_pos)
+        # Check range for axes which have finite ranges (for the axes we are moving)
+        # for full rotational axes moving freely is allowed
         for an in shift.keys():
             rng = self.axes[an].range
             p = target_pos[an]
-            if not rng[0] <= p <= rng[1]:
-                raise ValueError("Relative move would cause axis %s out of bound (%g m)" % (an, p))
+            if an != "rz":  # only rz has full rotational support and shouldn't be checked
+                if not rng[0] <= p <= rng[1]:
+                    raise ValueError("Relative move would cause axis %s out of bound (%g m)" % (an, p))
```

Comment:
Yeah but we should drop "rx" since that is not an axis which supports full rotation, only 0-90°.
So then line 612 would become:
if an in ("rz") and util.almost_equal(rng[1] - rng[0], 2 * math.pi)
But this feels like a double check right?

Reply:
Would in this case not only:

if an != "rz":

suffice?

Reply:
No, keep "rx". Let's be open-minded. The first SEM we are targetting doesn't support full rotation, but that doesn't mean the next ones will not.